### ISSUE TITLE
"defineConstant" doesn't seem to be working in a pyrightconfig.json

### ISSUE THREAD

**Issue by rchiodo** [microsoft/pyright#10372]
> Howerver, in VSCode, my "defineConstant" does not work. For example, 
> ```json
> "defineConstant": {
>     "__dev__": true,
> },
> ```
> But `__dev__` in my code is still warnning in VSCode. View Problems:
> "\_\_dev_\_\" is not defined Pylance[(reportUndefinedVariable)](https://github.com/microsoft/pyright/blob/main/docs/configuration.md#reportUndefinedVariable)
> 
>  

 _Originally posted by @GosickGeassGate in [#4301](https://github.com/microsoft/pylance-release/issues/4301#issuecomment-2830492380)_

**Comment by GosickGeassGate** [comment]
您好，我已经收到了您的邮件，祝您生活愉快。

**Comment by rchiodo** [comment]
I can reproduce with something like so:

```json
"defineConstant": {
    "DEBUG": true
}
```

```python
if DEBUG: # 
    print("Debug")
```

![Image](https://github.com/user-attachments/assets/8e2346ce-b5fb-480e-aaa6-c561ac33106e)

**Comment by rchiodo** [comment]
Hmm, maybe the documentation could be updated? I can see how it can be read as it actually assumes the constant is defined.

- defineConstant [map of constants to values (boolean or string), optional]: Set of identifiers that should be assumed to contain a constant value wherever used within this program. For example, { "DEBUG": true } indicates that pyright should assume that the identifier DEBUG will always be equal to True. If this identifier is used within a conditional expression (such as if not DEBUG:) pyright will use the indicated value to determine whether the guarded block is reachable or not. Member expressions that reference one of these constants (e.g. my_module.DEBUG) are also supported.
